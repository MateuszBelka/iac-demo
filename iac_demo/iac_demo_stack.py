from aws_cdk import (
    Stack,
)
from constructs import Construct

from .constructs.org_asg import ASG
from .constructs.org_standards.org_no_publicip import NoPublicIP


class IacDemoStack(Stack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        ASG(self, construct_id="AutoScalingGroup")

        NoPublicIP(self, construct_id="NoPublicIPConfigRule")
