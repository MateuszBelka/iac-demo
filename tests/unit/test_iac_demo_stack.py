import aws_cdk as core
import aws_cdk.assertions as assertions

from iac_demo.iac_demo_stack import IacDemoStack


def test_snapshot(snapshot):
    app = core.App()
    stack = IacDemoStack(app, "org-iac-demo")
    template = assertions.Template.from_stack(stack)
    assert template.to_json() == snapshot
