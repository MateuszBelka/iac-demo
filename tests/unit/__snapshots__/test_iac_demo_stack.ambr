# serializer version: 1
# name: test_snapshot
  dict({
    'Parameters': dict({
      'BootstrapVersion': dict({
        'Default': '/cdk-bootstrap/hnb659fds/version',
        'Description': 'Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]',
        'Type': 'AWS::SSM::Parameter::Value<String>',
      }),
    }),
    'Rules': dict({
      'CheckBootstrapVersion': dict({
        'Assertions': list([
          dict({
            'Assert': dict({
              'Fn::Not': list([
                dict({
                  'Fn::Contains': list([
                    list([
                      '1',
                      '2',
                      '3',
                      '4',
                      '5',
                    ]),
                    dict({
                      'Ref': 'BootstrapVersion',
                    }),
                  ]),
                }),
              ]),
            }),
            'AssertDescription': "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
          }),
        ]),
      }),
    }),
  })
# ---
